public/v4/apps/gotenberg.yml

```diff
@@ -0,0 +1,310 @@
+                - ARG PROMETHEUS_DISABLE_COLLECT
+                - ARG PROMETHEUS_DISABLE_ROUTE_LOGGING
+                - ARG LOG_FORMAT
+                - ARG LOG_LEVEL
+                - ARG GRACEFUL_SHUTDOWN_DURATION
+                - >-
+                    CMD gotenberg
+                    --api-port=$API_PORT
+                    --api-timeout=$API_TIMEOUT
+                    --api-root-path=$API_ROOT_PATH
+                    --api-trace-header=$API_TRACE_HEADER
+                    --api-disable-health-check-logging=$API_DISABLE_HEALTH_CHECK
+                    --chromium-allow-file-access-from-files=$CHROMIUM_ALLOW_FILE_ACCESS_FROM_FILES
+                    --chromium-allow-insecure-localhost=$CHROMIUM_ALLOW_INSECURE_LOCALHOST
+                    --chromium-allow-list=$CHROMIUM_ALLOW_LIST
+                    --chromium-deny-list=$CHROMIUM_DENY_LIST
+                    --chromium-ignore-certificate-errors=$CHROMIUM_IGNORE_CERTIFICATE_ERRORS
+                    --chromium-disable-web-security=$CHROMIUM_DISABLE_WEB_SECURITY
+                    --chromium-incognito=$CHROMIUM_INCOGNITO
+                    --chromium-host-resolver-rules=$CHROMIUM_HOST_RESOLVER_RULES
+                    --chromium-proxy-server=$CHROMIUM_PROXY_SERVER
+                    --chromium-disable-javascript=$CHROMIUM_DISABLE_JAVASCRIPT
+                    --chromium-disable-routes=$CHROMIUM_DISABLE_ROUTES
+                    --libreoffice-disable-routes=$LIBREOFFICE_DISABLE_ROUTES
+                    --uno-listener-start-timeout=$LIBREOFFICE_LISTENER_START_TIMEOUT
+                    --uno-listener-restart-threshold=$LIBREOFFICE_LISTENER_RESTART_THRESHOLD
+                    --pdfengines-engines=$PDFENGINES_ENGINES
+                    --pdfengines-disable-routes=$PDFENGINES_DISABLE_ROUTES
+                    --webhook-allow-list=$WEBHOOK_ALLOW_LIST
+                    --webhook-deny-list=$WEBHOOK_DENY_LIST
+                    --webhook-error-allow-list=$WEBHOOK_ERROR_ALLOW_LIST
+                    --webhook-error-deny-list=$WEBHOOK_ERROR_DENY_LIST
+                    --webhook-max-retry=$WEBHOOK_MAX_RETRY
+                    --webhook-retry-min-wait=$WEBHOOK_RETRY_MIN_WAIT
+                    --webhook-retry-max-wait=$WEBHOOK_RETRY_MAX_WAIT
+                    --webhook-client-timeout=$WEBHOOK_CLIENT_TIMEOUT
+                    --webhook-disable=$WEBHOOK_DISABLE
+                    --prometheus-collect-interval=$PROMETHEUS_COLLECT_INTERVAL
+                    --prometheus-namespace=$PROMETHEUS_NAMESPACE
+                    --prometheus-disable-collect=$PROMETHEUS_DISABLE_COLLECT
+                    --prometheus-disable-route-logging=$PROMETHEUS_DISABLE_ROUTE_LOGGING
+                    --log-format=$LOG_FORMAT
+                    --log-level=$LOG_LEVEL
+                    --gotenberg-graceful-shutdown-duration=$GRACEFUL_SHUTDOWN_DURATION
+        environment:
+            VERSION: $$cap_GOTENBERG_VERSION
```

Comment:
I don't think these environment variables (also `ARG`s in the dockerfileLines) are needed as your just passing the variables directly to the command line... I might be wrong, but maybe test it out without these and see if it works.

Merging anyways. But if it works without these, let's open another PR to clean it up. Thanks!

Reply:
I added the environment variables to be able to edit them later after deployment.

Regarding the `ARG`s in the `dockerfileLines` you are absolutely right, no need to define `ARG`s as we can just use `${var}` directly.

Thanks for the hints.

Since this PR is still open, I can edit it right here if you like.

Reply:
Ok, just tested.

When I remove the `ARG`s and I want to change an environment variable later after deployment, the change is not reflected since it just restarts the container with the environment variables from first creation during one-click-app deployment.

My approach with `ARG`s will recreate the container with the changed environment variables.
Therefore, I suggest accepting the PR as it is now, unless there are any other objections.

Reply:
Ah I see! Makes sense! Thanks for confirming that!